File: src/uva/Leetcode00024.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Leetcode00024 {
    public ListNode swapPairs(ListNode head) {
        if (head == null) {
            return null;
        }
        if (head.next != null) {
            ListNode second = head.next;
            head.next = swapPairs(head.next.next);
            second.next = head;
            return second;
        } else {
            return head;            
        }
    }
}

File: src/main/Leetcode00024.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */
class Solution {
    func swapPairs(head: ListNode?) -> ListNode? {
        var first : ListNode? = head
        var second : ListNode? = head?.next ?? nil
        var third : ListNode?
        var prehead = ListNode(0), previous : ListNode? = prehead
        if second == nil {
            return first
        }
        while first != nil && second != nil {
            third = second?.next
            
            first!.next = third
            second!.next = first
            previous?.next = second
            
            previous = first
            first = third
            second = third?.next
            
        }
        return prehead.next
    }
}
